app/stac_api/management/commands/update_asset_file_size.py

```diff
@@ -0,0 +1,59 @@
+import logging
+
+from botocore.exceptions import ClientError
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from stac_api.models import Asset
+from stac_api.models import CollectionAsset
+from stac_api.utils import AVAILABLE_S3_BUCKETS
+from stac_api.utils import CommandHandler
+from stac_api.utils import get_s3_client
+from stac_api.views.upload import SharedAssetUploadBase
+
+# path definition relative to the directory that contains manage.py
+DATADIR = settings.BASE_DIR / 'app/stac_api/sample_data/'
+logger = logging.getLogger(__name__)
+
+
+class Handler(CommandHandler):
+
+    def update(self):
+        self.print_success('running command to update file size')
+        s3 = get_s3_client(AVAILABLE_S3_BUCKETS.legacy)
+        bucket = settings.AWS_SETTINGS[AVAILABLE_S3_BUCKETS.legacy.name]['S3_BUCKET_NAME']
+
+        self.print_success('update file size for assets')
+        assets = Asset.objects.filter(file_size=0).all()
+        for asset in assets:
+            key = SharedAssetUploadBase.get_path(None, asset)
+            try:
+                file_size = s3.head_object(Bucket=bucket, Key=key)['ContentLength']
+                asset.file_size = file_size
+                asset.save()
+            except ClientError:
```

Comment:
this will fail for all assets that are stored on the managed bucket. I'd prepare the command to have two clients, legacy and managed and then check the correct bucket based on the info in the asset object